test/linter/test-consistency.js

```diff
@@ -217,16 +225,20 @@ class ConsistencyChecker {
   }
 
   /**
-   * @param {Identifier} data
-   * @return {data is Required<IdentifierMeta>}
+   * Checks if the data is a feature
+   *
+   * @param {Identifier} data The data to test
+   * @returns {boolean} If the data is a feature statement
```

Comment:
Both&nbsp;TypeScript’s&nbsp;JSDoc&nbsp;parser [and&nbsp;the&nbsp;ESLint&nbsp;JSDoc&nbsp;plugin](https://www.npmjs.com/package/eslint-plugin-jsdoc)’s [parser](https://github.com/jsdoctypeparser/jsdoctypeparser) consider&nbsp;this&nbsp;valid, and&nbsp;it’s&nbsp;used by&nbsp;TypeScript’s type&nbsp;checker&nbsp;to&nbsp;infer the&nbsp;type&nbsp;of&nbsp;`data` in&nbsp;an&nbsp;if‑block&nbsp;using the&nbsp;`isFeature`&nbsp;function.

```suggestion
   * @return {data is Required<IdentifierMeta>} If the data is a feature statement
```

Reply:
A couple of things:

1) In JSDoc, `@return` is a synonym for `@returns` according to JSDoc 3 (see https://jsdoc.app/tags-returns.html).  While both will work, we should pick only one and stick with it.  I'm in favor of `@returns` as a) it's the standard property name, b) the plugin's defaults prefer it, and c) it reads just a tiny bit better to me ("this function _returns_ the following data type").

2) The `Required` keyword is a TypeScript thing, not a JSDoc thing.  The closest applicable syntax would be a `!` before the data ("non-nullable"), however in my understanding, not specifying `null` or `undefined` as a possible return type makes it required anyways.

3) Actually, JSDoc does NOT allow for that syntax.  A quick test revealed a syntax error:
`221:0  warning  Syntax error in type: data is Required<IdentifierMeta>`
`221:0  warning  Syntax error in type: data is IdentifierMeta`

Reply:
Right, that’s&nbsp;because&nbsp;`jsdoctypeparser` still&nbsp;only&nbsp;has partial&nbsp;support for&nbsp;**TypeScript**&nbsp;types.

Until&nbsp;we&nbsp;add&nbsp;support for&nbsp;`param is Type`, I&nbsp;suggest you&nbsp;simply&nbsp;ignore that&nbsp;error.

See&nbsp;<https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50> for&nbsp;details.